lib/Sema/TypeChecker.h

```diff
@@ -556,6 +553,9 @@ class TypeChecker final {
   std::vector<AbstractClosureExpr *> ClosuresWithUncomputedCaptures;
 
 private:
+  ASTContext &Context;
+  DiagnosticEngine &Diags;
```

Comment:
How much work would it be to remove `Diags` altogether?

Reply:
@CodaFi Ah nice, there's only one use. I'll remove it future PR :)